src/server/apsystem.test.ts

```diff
@@ -215,6 +215,65 @@ test('ActivityPubSystem - List replies', async t => {
   t.deepEqual(collection.items, [activity.object])
 })
 
+test('ActivityPubSystem - Undo activity', async t => {
+  const store = newStore()
+  const hookSystem = new HookSystem(store, mockFetch)
+  const aps = new ActivityPubSystem('http://localhost', store, mockModCheck, hookSystem, mockServer.log)
+
+  const actorMention = '@user1@example.com'
+  const inReplyTo = 'https://example.com/note2'
+
+  // Sample data for the tests
+  const activity: APActivity = {
+    '@context': 'https://www.w3.org/ns/activitystreams',
+    type: 'Create',
+    published: new Date().toISOString(),
+    actor: 'https://example.com/user1',
+    object: {
+      type: 'Note',
+      published: new Date().toISOString(),
+      content: 'Hello world',
+      id: 'https://example.com/note1',
+      inReplyTo,
+      attributedTo: 'https://example.com/user1'
+    },
+    id: 'https://example.com/activity1'
+  }
+
+  const undoActivity: APActivity = {
+    '@context': 'https://www.w3.org/ns/activitystreams',
+    type: 'Undo',
+    actor: 'https://example.com/user1',
+    object: {
+      id: activity.id,
+      type: activity.type
+    },
+    id: 'https://example.com/undo1'
+  }
+
+  await store.forActor(actorMention).inbox.add(activity)
+  await aps.approveActivity(actorMention, activity.id as string)
+
+  // Log to verify the activity is in the store
```

Comment:
Ensure there's a check that the original activity is actually removed or marked as undone in some way. Currently, the test only checks if the undo activity is stored.

Reply:
Great, thanks! I accepted the changes and now I'm working on the updates and on this